ui/test/unit/appointments/controllers/manage/calendar/appointmentsCalendarViewController.spec.js

```diff
@@ -38,11 +39,12 @@ describe('AppointmentsCalendarViewController', function () {
 
     it('should get appointments for date', function () {
         var viewDate = new Date('1970-01-01T11:30:00.000Z');
+		state.params = {doFetchAppointmentsData: true};
         appointmentsService.getAllAppointments.and.returnValue(specUtil.simplePromise({data: []}));
         scope.getAppointmentsForDate(viewDate).then(function () {
             expect(scope.shouldReload).toBeFalsy();
             expect(stateParams.viewDate).toEqual(viewDate);
-        });
+        }); 
```

Comment:
Shouldn't you check here that `doFetchAppointmentsData` has become `false` after one round of calls?

Reply:
If yes this should be done for all tests for which the same verification may apply.

Reply:
Is it possible to show that when the appropriate state change occurs `getAllAppointments` is not called again? That's the test we are really after.

Reply:
Its possible I guess.Let me try it out